File: src/main/java/tech/bobliu/assignment08/service/ThreadService.java
package tech.bobliu.assignment08.service;

import tech.bobliu.assignment08.model.Post;
import tech.bobliu.assignment08.model.Thread;
import tech.bobliu.assignment08.model.User;

import java.util.List;

public interface ThreadService {
    List<Thread> getAllThreads();
    Thread getThread(Long id);
    Thread createThread(String title, String content, User sender);
    Post reply(Long threadId, String content, User sender, Long replyToId);
}

File: src/main/kotlin/tech/bobliu/assignment08/service/ThreadService.kt
package tech.bobliu.assignment08.service

import tech.bobliu.assignment08.model.Post
import tech.bobliu.assignment08.model.Thread
import tech.bobliu.assignment08.model.User

interface ThreadService {
    fun getAllThreads(): List<Thread>
    fun getThread(id: Long): Thread?
    fun createThread(title: String, content: String, sender: User): Thread
    fun reply(threadId: Long, content: String, sender: User, replyToId: Long? = null): Post
}
